File: js/tree_sitter_demo.js

const Parser = require('web-tree-sitter');

(async () => {
  await Parser.init();
  const parser = new Parser();
  const Lang = await Parser.Language.load('tree-sitter-q.wasm');
  parser.setLanguage(Lang);
  const tree = parser.parse('factorial:{$[x<2;1;x*.z.s x-1]}');
  console.log(tree.rootNode.toString());
  console.log(tree.rootNode);
  console.log(tree.rootNode.type);
  //console.log(JSON.stringify(tree.rootNode, null, "  "));

  console.log(JSON.stringify(tree.rootNode, function(key, value) {
      if (typeof value === 'function') {
        return value.toString();
      } else {
        return value;
      }
  }, "  "));

  const expr = tree.rootNode.child(0);
  console.log(expr.type);
  console.log(expr);
})();

File: python/tree_sitter_demo.py
from tree_sitter import Language, Parser


def main():
    print("Hello World!")

    Language.build_library(
        # Store the library in the `build` directory
        'build/my-languages.so',

        # Include one or more languages
        [
            '../vendor/tree-sitter-python',
        ]
    )

    PY_LANGUAGE = Language('build/my-languages.so', 'python')

    parser = Parser()
    parser.set_language(PY_LANGUAGE)

    # tree = parser.parse(bytes("""
    # def foo():
    #     if bar:
    #         baz()
    # """, "utf8"))

    tree = parser.parse(bytes("""
        1+2
        """, "utf8"))

    root_node = tree.root_node
    print(root_node.sexp())


if __name__ == "__main__":
    main()
